src/File.php

```diff
@@ -251,6 +256,195 @@ public function __construct($url, $timeout = 10, $redirects = 5, $headers = null
                 $this->success = false;
             }
         }
+
+        foreach ($this->headers as $name => $header) {
+            if (! is_array($header)) {
+                if (strpos($header, ',') === false) {
+                    $header = [$header];
+                } else {
+                    $header_lines = explode(',', $header);
+                    $header = [];
+                    foreach ($header_lines as $header_line) {
+                        $header[] = trim($header_line);
+                    }
+                }
+            }
+
+            $this->parsed_headers[strtolower($name)] = $header;
+        }
```

Comment:
Having the headers parsed in the __construct() method may be the fastest method, but has some drawbacks in BC. I will rewrite this so it will work without the `$parsed_headers` attribute.